packages/VSNumberCompare/index.js

```diff
@@ -129,19 +62,107 @@ function buildLegend(values, contrast, styles) {
       commonStyles.push(styles.contrast);
     }
 
-    legend.push(<Text key={ `${ value.unit }` } style={ [ ...commonStyles ] }>{ value.unit.toUpperCase() }</Text>);
-    legend.push(<Text key={ `${ value.unit }:` } style={ [ ...commonStyles, styles.legendSpacing ] }>{ ':' }</Text>);
+    legend.push(<Text key={ `${ value.unit }` } style={ [ ...commonStyles ] }>{value.unit.toUpperCase()}</Text>);
+    legend.push(<Text key={ `${ value.unit }:` } style={ [ ...commonStyles, styles.legendSpacing ] }>{':'}</Text>);
 
-    return value; // Make linter happy
+    return value;
   });
 
-  // Remove trailing colon
   legend.pop();
 
   return legend;
 }
 
-VSNumberCompare.propTypes = {
+class UI extends React.Component {
+  static get defaultProps() {
+    return {
+      title: 'Defaults',
+      values: [
+        {
+          unit: 'defs',
+          value: 0,
+          color: 'black',
+        },
+      ],
+      legendPosition: 'bottom',
+      contrast: false,
+      fontMain: 'TradeGothicBold',
+      fontSecondary: 'TradeGothicThin',
+      fontSize: 72,
+      fontRatio: 4,
+      fontMinimum: 16,
+      numberSpacing: 1,
+      legendSpacing: 2,
+    };
+  }
+
+  render() {
+    const {
+      fontSize,
+      fontRatio,
+      fontMinimum,
+      fontMain,
+      fontSecondary,
+      numberSpacing,
+      legendSpacing,
+      contrast,
+      legendPosition,
+      title,
+      values,
+    } = this.props;
+
+    const SMALL_FONT = fontSize / fontRatio > fontMinimum ? fontSize / fontRatio : fontMinimum;
+
+    styles.value = {
```

Comment:
Why overwrite `styles` instead of creating a new `StyleSheet` called dynamicStyles or something like that?

Reply:
I'm not sure what we mean by this, what's an example of this? Do we mean doing `StyleSheet.create` on render of the component?